generator/src/lib.rs

```diff
@@ -17,6 +17,21 @@ use std::fmt::Display;
 use std::path::Path;
 use syn::Ident;
 
+macro_rules! match_variant {
```

Comment:
I wasn't really sure what to call this... other names I thought of were `partial_match` or `extract_variant`.